irc/client.go

```diff
@@ -2045,3 +2050,53 @@ func (client *Client) sendPush(endpoint string, keys webpush.Keys, urgency webpu
 	}
 	return err
 }
+
+func (session *Session) MetadataSubscriptions() []string {
```

Comment:
This doesn't effectively synchronize anything since you're returning `metadataSubscriptions` to code that no longer holds the lock. You want something like `HasMetadataSubscription(sub string) bool` that does a `slices.Contains` check while holding the read lock.